src/routes/articles.js

```diff
@@ -0,0 +1,104 @@
+import express from "express";
+import { PrismaClient } from "@prisma/client";
+import { assert } from "superstruct";
+import asyncHandler from "../middleWares/errorHandler.js";
+import { CreateArticle, PatchArticle } from "../middleWares/structs.js";
+
+const prisma = new PrismaClient();
+
+const articleRoute = express.Router();
+
+articleRoute
+  .route("/detail")
+  .get(asyncHandler(async (req, res) => {
+      const { offset = 0, limit = 10, order = "newest", search = "" } = req.query;
+      let orderBy;
+      switch (order) {
+        case "oldest":
+          orderBy = { createdAt: "asc" };
+          break;
+        case "newest":
+        default:
+          orderBy = { createdAt: "desc" };
+      }
+      const articles = await prisma.article.findMany({
+        where: {
+          OR: [
+            { title: { contains: search, mode: "insensitive" } },
+            { content: { contains: search, mode: "insensitive" } },
+          ],
+        },
+        orderBy,
+        skip: parseInt(offset),
+        take: parseInt(limit),
+      });
+      res.status(200).send(articles);
+    }))
+
+articleRoute
+  .route("/")
+  .get(asyncHandler(async (req, res) => {
+      const { offset = 0, limit = 10, order = "newest", search = "" } = req.query;
+      let orderBy;
+      switch (order) {
+        case "oldest":
+          orderBy = { createdAt: "asc" };
+          break;
+        case "newest":
+        default:
+          orderBy = { createdAt: "desc" };
+      }
+      const articles = await prisma.article.findMany({
+        where: {
+          OR: [
+            { title: { contains: search, mode: "insensitive" } },
+            { content: { contains: search, mode: "insensitive" } },
+          ],
+        },
+        select: {
+          id: true,
+          title: true,
+          content: true,
+          createdAt: true,
+        },
+        orderBy,
+        skip: parseInt(offset),
+        take: parseInt(limit),
+      });
+      res.status(200).send(articles);
+    }))
+  .post(asyncHandler(async (req, res) => {
+      assert(req.body,CreateArticle);
+      const articles = await prisma.article.create({
+        data: req.body,
+      });
+      res.status(201).send(articles);
+    }))
+
+articleRoute
+  .route("/:id")
+  .get(asyncHandler(async (req, res) => {
+      const { id } = req.params;
+      const articles = await prisma.article.findUnique({
+        where: { id },
+      });
+      res.status(201).send(articles);
+    }))
+  .patch(asyncHandler(async (req, res) => {
+      assert(req.body,PatchArticle);
+      const { id } = req.params;
+      const articles = await prisma.article.update({
+        where: { id },
+        data: req.body,
+      });
+      res.status(201).send(articles);
+    }))
+  .delete(asyncHandler(async (req, res) => {
+      const { id } = req.params;
+      await prisma.article.delete({
+        where: { id },
+      });
+      res.sendStatus(204);
```

Comment:
status 대신에 sendStatus를 사용한 이유가 있나요??(이건 제가 궁금해서 질문 드리는 겁니다~)
```suggestion
      res.status(204);
```

Reply:
delete 의 경우, 삭제한 이후에 별도의 제공할 수 있는 데이터(POST - 생성된 데이터와 Status, Patch -수정된 데이터와 Status) 가 없이 Stats만 나오기 때문에 staru만 확인하기 위해서 sendStatus를 사용했습니다 !

차라리 명확한 확인을 위해서 status(204).send({ message: 삭제 완료! }) 를 사용하는 것이 나을까요?